lib/msal-browser/src/interaction_client/SilentRefreshClient.ts

```diff
@@ -14,6 +14,11 @@ export class SilentRefreshClient extends StandardInteractionClient {
      * @param request
      */
     async acquireToken(request: CommonSilentFlowRequest): Promise<AuthenticationResult> {
+        if (request.silentTokenRetrievalStrategy === SilentTokenRetrievalStrategy.NetworkOnly) {
+            this.logger.error("SilentTokenRetrievalStrategy is set to NetworkOnly, not attempting to acquire refresh token");
+            throw new ServerError(BrowserConstants.INVALID_GRANT_ERROR);
```

Comment:
Why would this be an existing server error and not something new?

Reply:
Because there is downstream logic that looks for an invalid grant error already.

Reply:
(which could be updated to look for a second type of error, if we wanted.)

Reply:
The protocol says interactive auth for both invalid_grant and interaction_required errors should be a retry, reference [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow). 

I recently had a conversation with someone from CCS concluding:

1. **For requests with authz code:**
a. `interaction_required` will return `InteractionRequiredAuthError` 
b. `invalid_grant` will return errors to client

2. **For requests with RT:**
a. `interaction_required `will return `InteractionRequiredAuthError` 
b.  `invalid_grant` will silently retry /authorize, and will return errors to client in case of errors

There are two things we may need to address (@Robbie-Microsoft  please check this):
- Are we handling 1.b correctly?
- With these new options, should we introduce MSAL JS specific errors that do not fall under the above protocol sub cases due to user preference (not wanting to use RT etc)?